rust/azure_iot_operations_mqtt/src/azure_mqtt_adapter.rs

```diff
@@ -256,16 +272,24 @@ impl MqttConnectionSettings {
             publish_qos1_qos2_queue_size,
         };
 
-        let keep_alive = azure_mqtt::packet::KeepAlive::Duration(
-            #[allow(clippy::cast_possible_truncation)] // Easier to do it this way
-            NonZeroU16::new(self.keep_alive.as_secs() as u16).ok_or_else(|| {
+        let ping_after =
+            NonZeroU16::new(u16::try_from(self.keep_alive.as_secs()).map_err(|e| {
                 ConnectionSettingsAdapterError {
                     msg: "cannot convert keep_alive to NonZeroU16".to_string(),
-                    field: ConnectionSettingsField::UseTls(self.use_tls),
-                    source: None,
+                    field: ConnectionSettingsField::KeepAlive(self.keep_alive),
+                    source: Some(Box::new(e)),
                 }
-            })?,
-        );
+            })?)
+            .ok_or(ConnectionSettingsAdapterError {
+                msg: "keep_alive must be greater than zero".to_string(),
+                field: ConnectionSettingsField::KeepAlive(self.keep_alive),
+                source: None,
+            })?;
+
+        let keep_alive = azure_mqtt::client::KeepAliveConfig::Duration {
+            ping_after,
+            response_timeout: Duration::from_secs(2),
```

Comment:
Where does this 2 come from?

Reply:
Arbitrary - we can add a setting in Session Options to pipe it through later if we care, but this was never something we configured before.